### ISSUE TITLE
FalsePositive - NoSemicolons: Empty Enum with init

### ISSUE THREAD

**Issue by hfhbd** [detekt/detekt#8765]
## Expected Behavior
No positive

## Observed Behavior
False positive `NoSemicolons` error that cause a compiler error when `autoCorrect` is enabled.

## Steps to Reproduce
```kotlin
enum class Foo {
  ;
  init {
    error("This class will be generated and is only provided for compiling.")
  }
}
```

## Context
While an empty enum class is very rare, I want to generate an enum class via a code generator, but I need to provide a stub for compiling, so I used an empty enum.

## Your Environment

* Version of detekt used: 1.23.8
* Version of Gradle used (if applicable): 9.1.0
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: macOS 26.0.1
* Link to your project (if it's a public repository):

**Comment by 3flex** [comment]
This is a ktlint rule (detekt just provides a wrapper) so likely one to be raised with the ktlint project.

You could confirm by running ktlint directly against that code.

**Comment by hfhbd** [comment]
Oh thanks, will move the issue.

**Comment by 3flex** [comment]
Actually I'll take that back, it's been raised before but couldn't be replicated - let me know how you go.

https://github.com/detekt/detekt/issues/6615
https://github.com/pinterest/ktlint/issues/2351#issuecomment-1808976853

**Comment by atulgpt** [comment]
hi @hfhbd can you create a simple reproducer for this issue? Not able to repro this issue either in TC or actual code
